### ISSUE TITLE
AUTOGRAD is not working on IOS

### ISSUE THREAD

**Issue by zzmicer** [pytorch/pytorch#86271]
### 🐛 Describe the bug

``Hello! 
I've built latest pytorch from sources with `BUILD_MOBILE_AUTOGRAD : ON` on ios. However, when I'm trying to reproduce the easiest autograd example, I get an error.
```c++
{
            torch::AutoGradMode enable_grad(true);
            
            auto x = torch::ones({2, 2}, torch::requires_grad());
            auto y = x + 2;
            y = y.sum();
            y.backward();
 }
 
```
ERROR_LOG: `element 0 of tensors does not require grad and does not have a grad_fn`

My initial task is to collect the gradients from forward pass in order to preprocess input(to perform forward pass again, and get more robust softmax scores).

Seems like autograd is not working.

### Versions

BUILD FLAGS
--   TORCH_VERSION         : 1.13.0
--   CAFFE2_VERSION        : 1.13.0
--   BUILD_CAFFE2          : OFF
--   BUILD_CAFFE2_OPS      : OFF
--   BUILD_STATIC_RUNTIME_BENCHMARK: OFF
--   BUILD_TENSOREXPR_BENCHMARK: OFF
--   BUILD_NVFUSER_BENCHMARK: OFF
--   BUILD_BINARY          : OFF
--   BUILD_CUSTOM_PROTOBUF : OFF
--     Protobuf compiler   : 
--     Protobuf includes   : 
--     Protobuf libraries  : 
--   BUILD_DOCS            : OFF
--   BUILD_PYTHON          : OFF
--   BUILD_SHARED_LIBS     : OFF
--   CAFFE2_USE_MSVC_STATIC_RUNTIME     : ON
--   BUILD_TEST            : OFF
--   BUILD_JNI             : OFF
--   BUILD_MOBILE_AUTOGRAD : ON
--   BUILD_LITE_INTERPRETER: ON
--   CROSS_COMPILING_MACOSX : 
--   INTERN_BUILD_MOBILE   : ON
--   TRACING_BASED         : OFF
--   USE_BLAS              : OFF
--   USE_LAPACK            : 0
--   USE_ASAN              : OFF
--   USE_CPP_CODE_COVERAGE : OFF
--   USE_CUDA              : OFF
--   USE_ROCM              : OFF
--   USE_EIGEN_FOR_BLAS    : ON
--   USE_FBGEMM            : OFF
--     USE_FAKELOWP          : OFF
--   USE_KINETO            : OFF
--   USE_FFMPEG            : OFF
--   USE_GFLAGS            : OFF
--   USE_GLOG              : OFF
--   USE_LEVELDB           : OFF
--   USE_LITE_PROTO        : OFF
--   USE_LMDB              : OFF
--   USE_METAL             : ON
--   USE_PYTORCH_METAL     : OFF
--   USE_PYTORCH_METAL_EXPORT     : OFF
--   USE_MPS               : OFF
--   USE_FFTW              : OFF
--   USE_MKL               : 
--   USE_MKLDNN            : OFF
--   USE_UCC               : OFF
--   USE_ITT               : OFF
--   USE_NCCL              : OFF
--   USE_NNPACK            : OFF
--   USE_NUMPY             : OFF
--   USE_OBSERVERS         : OFF
--   USE_OPENCL            : OFF
--   USE_OPENCV            : OFF
--   USE_OPENMP            : OFF
--   USE_TBB               : OFF
--   USE_VULKAN            : OFF
--   USE_PROF              : OFF
--   USE_QNNPACK           : OFF
--   USE_PYTORCH_QNNPACK   : ON
--   USE_XNNPACK           : ON
--   USE_REDIS             : OFF
--   USE_ROCKSDB           : OFF
--   USE_ZMQ               : OFF
--   USE_DISTRIBUTED       : OFF
--   USE_DEPLOY           : OFF
--   Public Dependencies  : caffe2::Threads
--   Private Dependencies : pthreadpool;cpuinfo;pytorch_qnnpack;XNNPACK;fp16;fmt::fmt-header-only
--   USE_COREML_DELEGATE     : OFF
--   BUILD_LAZY_TS_BACKEND   : OFF

**Comment by albanD** [comment]
I'm afraid I don't know how the mobile build is handling this argument. We would need someone from the mobile side to help here cc @raziel who would be able to help on this?

**Comment by raziel** [comment]
@zzmicer have you tried with one of the prebuilt binaries? Just want to check whether this is an issue in the current code and/or the build process.

@kimishpatel @jmdetloff   do you see any obvious issues here?

**Comment by zzmicer** [comment]
Is there any other method to perform backward pass(or to get computation backward graph) in order to gather gradients w.r.t loss?

**Comment by kimishpatel** [comment]
@iseeyuan do you know how well `BUILD_MOBILE_AUTOGRAD : ON && BUILD_LITE_INTERPRETER: ON` is supported? From this https://github.com/pytorch/pytorch/blob/master/caffe2/CMakeLists.txt#L348, it does not seem to be the case. @zzmicer maybe something you can try out and see if it works?

**Comment by iseeyuan** [comment]
> @iseeyuan do you know how well `BUILD_MOBILE_AUTOGRAD : ON && BUILD_LITE_INTERPRETER: ON` is supported? From this https://github.com/pytorch/pytorch/blob/master/caffe2/CMakeLists.txt#L348, it does not seem to be the case. @zzmicer maybe something you can try out and see if it works?

To my understanding, it means building lite interpreter with training capability. @cccclai , could you confirm?

**Comment by zzmicer** [comment]
> To my understanding, it means building lite interpreter with training capability. @cccclai , could you confirm?

But actually I couldn't find any examples of on-device training :(

**Comment by cccclai** [comment]
I'm not sure this config..I don't think we support training on mobile for now?
